### ISSUE TITLE
Storybook a11y improvements

### ISSUE THREAD

**Issue by ling1726** [microsoft/fluentui#18065]
#17960  proposes to use Storybook as documentation for convergence/v9. Storybook can be considered accessible after an initial review by @jurokapsiar, but the experience is not ideal. If effort is available we can look into the following improvements in the future.

-  no logical structure of the page and/or no standardized skiplink (there is a shortcut but not a standard one) - at least we should add regions
- left side nav is not marked in any way, user does not understand that it could be a hierarchical strcture with expandable sections
- just tabbing to navigate through left side nav links makes it hard for the keyboard users to navigate
- by default shortcuts do not have modifiers, which might collide with screen reader keys - user needs to remap shortcuts to start using them
- color contrast for section headers is insufficient
- show/hide code button does not suggest that it expands a new section
- props table does not have a label

**Comment by gouttierre** [comment]
@ling1726 : Thanks for the suggestions!
